### PR TITLE
Extend server state transitions

### PR DESCRIPTION
**Pull request by stefanak-michal**
By description at https://neo4j.com/docs/bolt/current/bolt/message/#messages-route it can change server state into those.

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates.

You can view the updated docs at https://neo4j-docs-bolt-48.surge.sh

File: modules/ROOT/pages/appendix/version-4.adoc
Comment (robsdedude) [inline modules/ROOT/pages/appendix/version-4.adoc:1]: Changes look reasonable to me, can we please also apply them to the version 5 document?

Comment (robsdedude) [reply modules/ROOT/pages/appendix/version-4.adoc:1]: We could also consider adding `FAILED` `ROUTE` -> `IGNORED` `FAILED` to both.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-bolt-48.surge.sh

Updated pages:
- [Bolt message state transitions in version 4.x](https://neo4j-docs-bolt-48.surge.sh/bolt/1.0/appendix/version-4)

**Review by robsdedude — APPROVED** [review]
:five:

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-bolt-48.surge.sh

Updated pages:
- [Bolt message state transitions in version 4.x](https://neo4j-docs-bolt-48.surge.sh/bolt/1.0/appendix/version-4)
- [Bolt message state transitions in version 5.1<br>](https://neo4j-docs-bolt-48.surge.sh/bolt/1.0/appendix/version-5)

**Review by thelonelyvulpes — APPROVED** [review]

**Pull request merged**